### ISSUE TITLE
Crashing with drawing tablet

### ISSUE THREAD

**Issue by dmyger** [synfig/synfig#351]
**Synfig version & platform**:
Windows 10 Pro 64 bit
synfigstudio-1.3.2-20170408-64bit.exe 
Tablet XP-Pen G540

**Issue description**:
Synfig now crushed when the drawing pen only move in any area of screen, even if the program minimazed. It does not matter is tabled disabled in input devices settings or not.
Previous version in spite of works not correctly, but at least had not crashed on any manipulations with the pen on the same tablet.

Message from Windows event viewer:
_"Faulting application name: synfigstudio.exe, version: 0.0.0.0, time stamp: 0x58f8df29
Faulting module name: libgdk-3-0.dll, version: 3.22.2.0, time stamp: 0x58f8d723
Exception code: 0xc0000005
Fault offset: 0x00000000000419f4"_

**Comment by morevnaproject** [comment]
Hello! So, If I understand correctly, you do the following:
1. Start Synfig using mouse.
2. Make a move using pen  of tablet.
3. Synfig crashes.

Am I right?

**Comment by dmyger** [comment]
Exactly.

**Comment by morevnaproject** [comment]
Can you please test if your tablet works with latest Gimp or MyPaint?

https://github.com/mypaint/mypaint/releases/tag/v1.2.1
https://download.gimp.org/mirror/pub/gimp/v2.8/windows/gimp-2.8.20-setup.exe

**Comment by dmyger** [comment]
It works well at least with Gimp 2.8.18 and Krita 3.1.2. With Synfig 1.2.0 & 1.3.1  it works too, has issues, but no crash.
To test with MyPaint I need a couple days.

**Comment by blackwarthog** [comment]
Please, try to run following commands from console:

```
C:
cd "C:\Program Files\Synfig\bin"
set GDK_PIXBUF_MODULEDIR="../lib/gdk-pixbuf-2.0/2.10.0/loaders/"
gdk-pixbuf-query-loaders.exe --update-cache
```

And try to run Synfig Studio again.

**Comment by dmyger** [comment]
> gdk-pixbuf-query-loaders.exe --update-cache
This breaks the launch of Synfig completely
[synfigstudio_console_log.txt](https://github.com/synfig/synfig/files/954502/synfigstudio_console_log.txt)

**Comment by dmyger** [comment]
Reinstall version 1.3.2 - got the same crashes on pen move.

> latest Gimp or MyPaint

1) GIMP: While it disabled in Input Devices it works like mouse, but if I enable "Vi Pressure Stylus" (any screen or window) - it does not work. No any drawing, just stochastic movements, open menu and others random actions.

2) MyPaint: It draw with issues. At the moment of touching it draws a line from far upper-right side to the current point, then it drawing correctly, while the pen touches the tablet. But there is no information about the force of pressure.

It seems like it was in previous versions of Synfig, but any way there are not crashed.
So only Krita works correctly with pressure.

**Comment by morevnaproject** [comment]
I think this is because we use different GTK version in 1.3.2. We will try to rebuild against other version.

**Comment by morevnaproject** [comment]
@dmyger Can you please check if this file works for you? - https://icystar.com/downloads/experimental/SynfigStudio-1.3.3-2017.04.30-56018-win-64bit.exe

**Comment by dmyger** [comment]
Pos: at least it does not crashed.
Cons: drawing tablet does not work correctly. If it turned on in the input devices settings (Screen or Window). There is only cursor movements, but no drawing. More over, no any reaction in UI on click by pen.
If it turned off as input device, it works like mouse with no pressure info.

Console log file attached, if it could helps.
[synfig-1.3.3.txt](https://github.com/synfig/synfig/files/1004720/synfig-1.3.3.txt)

**Comment by morevnaproject** [comment]
Thank you fro testing! I am happy to know the crash is gone. 

Did the tablet pressure was working with 1.2.0 version of Synfig?

Judging from behavior in GIMP. I suspect XP-Pen tablets are not compatible with Wacom and thus didn't supported by Gtk correctly. I found similar reports in MyPaint bugtracker - https://github.com/mypaint/mypaint/issues/584#issuecomment-182333921

**Comment by dmyger** [comment]
I see that only Krita works correctly with this tablet, from free software what I have.
1.2.0 there is no pressure reaction for Draw tool and has some other issues: with first click it often goes to far right and up. Also intresting UI reaction on click: menu bar has no responce on click, but toolboxes works well.
Any way it seems that now it is a different issue.

**Comment by morevnaproject** [comment]
> I see that only Krita works correctly with this tablet, from free software what I have.

Krita devs got rid of tablet engine by QT and wrote their own with a purpose to support more tablets than QT does. ^__^

**Comment by morevnaproject** [comment]
Shipped with 1.3.3 - https://icystar.com/downloads/experimental/SynfigStudio-1.3.3-2017.04.30-56018-win-64bit.exe

**Comment by morevnaproject** [comment]
Shipped with 1.3.3 - https://morevnaproject.org/2017/05/22/synfig-studio-1-3-3-development/

**Comment by morevnaproject** [comment]
@dmyger You can try to contact [Anton Sibiryakov](https://vk.com/antonxppen), maybe he consider to collaborate with Gtk developers to bring support for XP-Pen tablets.
